src/core/credentials/transport/tls/grpc_tls_certificate_provider.h

```diff
@@ -208,6 +210,62 @@ class FileWatcherCertificateProvider final
   std::map<std::string, WatcherInfo> watcher_info_ ABSL_GUARDED_BY(mu_);
 };
 
+// Implements a provider that uses in-memory data that can be modified in a
+// thread-safe manner.
+class InMemoryCertificateProvider final : public grpc_tls_certificate_provider {
+ public:
+  static std::shared_ptr<InMemoryCertificateProvider> CreateCertificateProvider(
```

Comment:
We should not be using `shared_ptr<>` here.  The cert provider base class inherits from `RefCounted<>`; it's intended to be used with `RefCountedPtr<>`.

I don't think we need this factory function.  Callers can just do `MakeRefCounted<InMemoryCertificateProvider>(...)`.